### PR TITLE
Fix column removal in MSSQL

### LINKED ISSUE #3624: On Update to 1.4.0-rc2: Error on ALTER TABLE DROP COLUMN is_owner
- Gitea version (or commit ref): v1.3.2/3 => v1.4.0-rc2
- Git version: git version 2.7.4
- Operating system: Linux vserv24 4.4.0-112-generic #135-Ubuntu SMP Fri Jan 19 11:48:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description
Hi

I update with the following procedure:
```
sudo systemctl stop gitea
sudo su git
cd /home/git/gitea
mv gitea gitea.old
wget https://github.com/go-gitea/gitea/releases/download/v1.4.0-rc2/gitea-1.4.0-rc2-linux-amd64
mv gitea-1.4.0-rc2-linux-amd64 gitea
chmod +x gitea
./gitea web
```

The following logs are genereated, gitea does not start up.

./log/gitea.log:
```
2018/03/05 08:11:49 [I] Migration: remove is_owner, num_teams columns from org_user
2018/03/05 08:11:49 [...itea/routers/init.go:60 GlobalInit()] [E] Failed to initialize ORM engine: migrate: do migrate: DROP COLUMN org_user.is_owner, org_user.num_teams: mssql: Fehler bei ALTER TABLE DROP COLUMN is_owner, da mindestens ein Objekt auf diese Spalte zugreift.
```
Translation of the error (4922):  ALTER TABLE ALTER COLUMN is_owner failed because one or more objects access this column.

./log/xorm.log: `2018/03/05 08:03:58 [I] [SQL] ALTER TABLE org_user DROP COLUMN is_owner, num_teams`

I see there is a default value on this column. If I generate a DROP COLUMN script in sql-server it gernerates:

```
ALTER TABLE dbo.org_user
	DROP CONSTRAINT DF__org_user__is_own__797309D9
GO
ALTER TABLE dbo.org_user
	DROP COLUMN is_owner
GO
```

### PR DESCRIPTION
**Pull request by lafriks**
Fixes #3624

### CODE REVIEW

File: models/migrations/migrations.go [hunk 1]
```diff
@@ -217,6 +217,54 @@ Please try to upgrade to a lower version (>= v0.6.0) first, then upgrade to curr
...
+			cols += "`" + strings.ToLower(col) + "`"
+		}
+		sql := fmt.Sprintf("SELECT Name FROM SYS.DEFAULT_CONSTRAINTS WHERE PARENT_OBJECT_ID = OBJECT_ID('%[1]s') AND PARENT_COLUMN_ID IN (SELECT column_id FROM sys.columns WHERE lower(NAME) IN (%[2]s) AND object_id = OBJECT_ID('%[1]s'))",
+			tableName, strings.Replace(cols, "`", "'", -1))
+		constraints := make([]string, 0)
+		if err := x.SQL(sql).Find(&constraints); err != nil {
+			return fmt.Errorf("Find constraints: %v", err)
+		}
+		for _, constraint := range constraints {
+			if _, err := x.Exec(fmt.Sprintf("ALTER TABLE `%s` DROP CONSTRAINT `%s`", tableName, constraint)); err != nil {
```
Comment (lunny) [inline models/migrations/migrations.go]: How about use a transaction here?

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/migrations/migrations.go (hunk 1 above)
Comment (lafriks) [reply models/migrations/migrations.go]: Alter table is not transactional but I can add it

**Comment by lafriks** [comment]
@lunny done

**Review by lunny — APPROVED** [review]

**Comment by lafriks** [comment]
MSSQL alter table is actually transaction so I added transaction as well

**Review by appleboy — APPROVED** [review]

**Pull request merged**